### ISSUE TITLE
python setup.py install is broken

### ISSUE THREAD

**Issue by neworderofjamie** [genn-team/pynn_genn#31]
You currently need to run this twice to get a functioning build (as Jenkinsfile currently does). The reason is that the ``install`` command first runs ``egg_info``, then ``install_lib``, which runs ``build_py`` then ``build_ext``. ``build_py`` copies all Python sources into build directory (including those built by SWIG) and ``build_ext`` runs SWIG so, if you build in an existing location, bits of the build are out-dated and, if you build in a new location, bits of build are missing.

[Stackoverflow](https://stackoverflow.com/questions/12491328/python-distutils-not-include-the-swig-generated-module) suggests a fix which re-orders the order of ``install`` sub-commands so ``build_ext`` is done first. However that means that SWIG without libgenn (and, I'm guessing, other things) having been copied into the build directory meaning that it fails.

**Comment by agarwalmanvi** [comment]
Is this still open? I used python setup.py install and it exited without any errors

**Comment by neworderofjamie** [comment]
It is indeed. ``python setup.py install`` doesn't cause errors but, if you try and import PyGeNN, it won't work

**Comment by agarwalmanvi** [comment]
I can't seem to reproduce the issue. I imported both pynn_genn and pygenn in ipython successfully. Then I thought of running a test to confirm if everything works the way it should, so I looked at simple_network.py, but I ran into some issues with Nest.

**Comment by neworderofjamie** [comment]
This only happens in a brand new python environment i.e. a new virtualenv

**Comment by agarwalmanvi** [comment]
I created a fresh conda environment to see if there were any issues. This is what I did:
`conda create -n newenv python=3.7 numpy` (numpy is a dependency for PyGeNN)
Built PyGeNN from source (https://github.com/genn-team/genn/tree/master/pygenn) with `python setup.py develop`
Then installed PyNN-GeNN with `python setup.py install`
Again, imported `pynn_genn` and `pygenn` into ipython, no problems. Can you recreate this and see if this also works for you?

**Comment by neworderofjamie** [comment]
Ahh, *now* I think I see the problem. This issue is really about PyGeNN rather than PYNNGENN. If you install that with install rather than develop with a fresh checkout into a new virtualenv, you _should_ see the problem

**Comment by agarwalmanvi** [comment]
Nope, redid all of this with a fresh environment, replacing `develop` with `install` for PyGeNN. Still no problem importing both `pygenn` and `pynn_genn`...

**Comment by neworderofjamie** [comment]
Did you run setup from a clean install of PyGeNN? If you've built it before in the same directory, the problem doesn't occur

**Comment by agarwalmanvi** [comment]
Indeed this time I was able to get the following error: `No module named 'pygenn.genn_wrapper.genn_wrapper'`. But this error doesn't get resolved by running `python setup.py install` for PyNN-GeNN a second time.

**Comment by neworderofjamie** [comment]
You need to run ``python setup.py install`` for PyGeNN (not PyNN GeNN) a second time as that is the module with the problem (perhaps this bug needs moving...)
